clang/lib/Driver/ToolChain.cpp

```diff
@@ -974,8 +974,11 @@ std::string ToolChain::GetLinkerPath(bool *LinkerIsLLD) const {
       if (llvm::sys::path::parent_path(Path).empty())
         Path = GetProgramPath(A->getValue());
       if (llvm::sys::fs::can_execute(Path)) {
+        SmallString<1024> RealPath;
+        if (llvm::sys::fs::real_path(Path, RealPath))
```

Comment:
We should not make an extra syscall when `UseLinker == "lld"`

Reply:
Fixed